src/com/owncloud/android/lib/common/operations/RemoteOperation.java

```diff
@@ -25,78 +25,104 @@
 import com.owncloud.android.lib.common.OwnCloudClientManagerFactory;
-import com.owncloud.android.lib.common.OwnCloudCredentials;
 import com.owncloud.android.lib.common.accounts.AccountUtils;
-import com.owncloud.android.lib.common.operations.RemoteOperationResult.ResultCode;
 import com.owncloud.android.lib.common.utils.Log_OC;
 
 import java.io.IOException;
 
 
 /**
  * Operation which execution involves one or several interactions with an ownCloud server.
- * 
+ *
  * Provides methods to execute the operation both synchronously or asynchronously.
- * 
- * @author David A. Velasco 
+ *
+ * @author David A. Velasco
+ * @author David González Verdugo
  */
 public abstract class RemoteOperation implements Runnable {
-	
+
     private static final String TAG = RemoteOperation.class.getSimpleName();
 
-    /** OCS API header name */
+    /**
+     * OCS API header name
+     */
     public static final String OCS_API_HEADER = "OCS-APIREQUEST";
 
-    /** OCS API header value */
+    /**
+     * OCS API header value
+     */
     public static final String OCS_API_HEADER_VALUE = "true";
 
-    /** ownCloud account in the remote ownCloud server to operate */
+    /**
+     * ownCloud account in the remote ownCloud server to operate
+     */
     private Account mAccount = null;
-    
-    /** Android Application context */
+
+    /**
+     * Android Application context
+     */
     private Context mContext = null;
-    
-	/** Object to interact with the remote server */
-	private OwnCloudClient mClient = null;
-	
-	/** Callback object to notify about the execution of the remote operation */
-	private OnRemoteOperationListener mListener = null;
-	
-	/** Handler to the thread where mListener methods will be called */
-	private Handler mListenerHandler = null;
-
-	/** Activity */
-    private Activity mCallerActivity;
-
-	
-	/**
-	 *  Abstract method to implement the operation in derived classes.
-	 */
-	protected abstract RemoteOperationResult run(OwnCloudClient client);
-	
+
+    /**
+     * Object to interact with the remote server
+     */
+    private OwnCloudClient mClient = null;
+
+    /**
+     * Callback object to notify about the execution of the remote operation
+     */
+    private OnRemoteOperationListener mListener = null;
+
+    /**
+     * Handler to the thread where mListener methods will be called
+     */
+    private Handler mListenerHandler = null;
+
+    /**
+     * When 'true', the operation tries to silently refresh credentials if fails due to lack of authorization.
+     *
+     * Valid for 'execute methods' receiving an {@link Account} instance as parameter, but not for those
+     * receiving an {@link OwnCloudClient}. This is, valid for:
+     *  -- {@link RemoteOperation#execute(Account, Context)}
+     *  -- {@link RemoteOperation#execute(Account, Context, OnRemoteOperationListener, Handler)}
+     */
+    private boolean mSilentRefreshOfAccountCredentials = false;
```

Comment:
> We set it as false when methods with no account are executed

Right.

> and as true with methods that have an account available are executed.

No. This set is not done in the object itself, it is done externally, in our app. Other apps using the library could decide not do it, and in that case the refresh wouldn't be done at all.